File: bot.py
import logging
import openai
from telegram import Update
from telegram.ext import ApplicationBuilder, CommandHandler, MessageHandler, filters, ContextTypes

# ВСТАВЬ СВОИ КЛЮЧИ
TELEGRAM_BOT_TOKEN = "7865133019:AAEp6SAXei7l0TSF3fAnaVn_9FPrKURbWe0"
OPENAI_API_KEY = "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"

# Настройка GPT
openai.api_key = OPENAI_API_KEY
GPT_MODEL = "gpt-3.5-turbo"  # или "gpt-4o", если доступен

# Генерация ответа от GPT
async def ask_gpt(prompt: str) -> str:
    try:
        response = openai.ChatCompletion.create(
            model=GPT_MODEL,
            messages=[{"role": "user", "content": prompt}]
        )
        return response.choices[0].message.content.strip()
    except Exception as e:
        logging.error(f"Ошибка OpenAI: {e}")
        return "⚠️ Произошла ошибка при обращении к GPT."

# Команда /start
async def start(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await update.message.reply_text("Привет! Я GPT-бот. Напиши мне что-нибудь — я отвечу 🤖")

# Обработка сообщений
async def handle_message(update: Update, context: ContextTypes.DEFAULT_TYPE):
    user_message = update.message.text
    await update.chat.send_action("typing")
    response = await ask_gpt(user_message)
    await update.message.reply_text(response)

# Запуск бота
def main():
    logging.basicConfig(level=logging.INFO)
    app = ApplicationBuilder().token(TELEGRAM_BOT_TOKEN).build()

    app.add_handler(CommandHandler("start", start))
    app.add_handler(MessageHandler(filters.TEXT & ~filters.COMMAND, handle_message))

    print("Бот запущен.")
    app.run_polling()

if __name__ == "__main__":
    main()

File: bot.js
require('dotenv').config();
const { Telegraf } = require('telegraf');
const axios = require('axios');

const bot = new Telegraf(process.env.BOT_TOKEN);
const openaiApiKey = process.env.OPENAI_API_KEY;

bot.start((ctx) => {
  ctx.reply('Привет, это AI managed 🤖\n\nНапиши:\n\nНиша: ...\nЦА: ...\nПредложение: ...\n\nЯ сгенерирую 2 оффера.');
});

bot.on('text', async (ctx) => {
  const input = ctx.message.text;

  const prompt = `
Ты — AI-ассистент для создания продающих офферов.
Сгенерируй 2 оффера по шаблону:

🎯 Оффер для ниши: [указать]
1. 💡 Что предлагаешь
2. 🔥 Почему это важно
3. 🧠 Кто ты
4. 🚀 Призыв к действию

+ Варианты заголовков и CTA

Вводные:
${input}
`;

  try {
    const response = await axios.post(
      'https://api.openai.com/v1/chat/completions',
      {
        model: 'gpt-4o',
        messages: [{ role: 'user', content: prompt }],
        temperature: 0.7
      },
      {
        headers: {
          'Authorization': `Bearer ${openaiApiKey}`,
          'Content-Type': 'application/json'
        }
      }
    );

    const result = response.data.choices[0].message.content;
    ctx.reply(result);
  } catch (error) {
    console.error(error.response?.data || error.message);
    ctx.reply('⚠️ Ошибка. Проверь OpenAI ключ.');
  }
});

bot.launch();
console.log('🤖 AI managed bot запущен');
